File: app/src/main/java/com/bigfriedicecream/recipes/interfaces/IRecipeListContract.java
package com.bigfriedicecream.recipes.interfaces;

import android.content.Context;

import com.bigfriedicecream.recipes.models.RecipeDataModel;

import java.util.Map;

public interface IRecipeListContract {

    interface View {
        void render(Map<String, RecipeDataModel> recipeList);
    }

    interface Presenter {
        void load(Context context);
    }
}

File: Code/app/src/main/java/com/bigfriedicecream/recipes/interfaces/IRecipeListContract.kt
package com.bigfriedicecream.recipes.interfaces

import android.content.Context

import com.bigfriedicecream.recipes.models.RecipeDataModel

interface IRecipeListContract {

    interface View {
        fun render(recipeList: Map<String, RecipeDataModel>)
    }

    interface Presenter {
        fun start(c:Context)
        fun stop()
    }
}
